tests/js/plugin/enhancedanchoring-test.coffee

```diff
@@ -59,178 +36,212 @@ describe 'Annotator.Plugin.EnhancedAnchoring', ->
-      am = createAnchoringManager()
-      ann1 = createTestAnnotation "a1", 2
-      ann2 = createTestAnnotation "a2"
-      ann3 = createTestAnnotation "a3"
-      anchor11 = am.createAnchor(ann1, ann1.target[0]).result
-      anchor12 = am.createAnchor(ann1, ann1.target[1]).result
-      anchor2 = am.createAnchor(ann2, ann2.target).result
-      anchor3 = am.createAnchor(ann3, ann2.target).result
-
-      anchors = am.getAnchors [ann1, ann2]
-
-      assert.isArray anchors
-      assert.include anchors, anchor11
-      assert.include anchors, anchor12
-      assert.include anchors, anchor2
-      assert.notInclude anchors, anchor3
-
-  describe 'getHighlights', ->
-    it 'returns an empty array by default', ->
-      am = createAnchoringManager()
-      hls = am.getHighlights()
-      assert.isArray hls
-      assert.equal hls.length, 0
-
-    it 'returns all the highlights', ->
-      am = createAnchoringManager()
-      ann1 = createTestAnnotation "a1", 2
-      ann2 = createTestAnnotation "a2"
-      ann3 = createTestAnnotation "a3"
-      anchor11 = am.createAnchor(ann1, ann1.target[0]).result
-      anchor12 = am.createAnchor(ann1, ann1.target[1]).result
-      anchor2 = am.createAnchor(ann2, ann2.target).result
-      anchor3 = am.createAnchor(ann3, ann2.target).result
-
-      hls = am.getHighlights()
-
-      assert.isArray hls
-      assert.include hls, anchor11.highlight[anchor11.startPage]
-      assert.include hls, anchor12.highlight[anchor12.startPage]
-      assert.include hls, anchor2.highlight[anchor2.startPage]
-      assert.include hls, anchor3.highlight[anchor3.startPage]
-
-    it 'returns the highlights belonging to a set of annotations', ->
-      am = createAnchoringManager()
-      ann1 = createTestAnnotation "a1", 2
-      ann2 = createTestAnnotation "a2"
-      ann3 = createTestAnnotation "a3"
-      anchor11 = am.createAnchor(ann1, ann1.target[0]).result
-      anchor12 = am.createAnchor(ann1, ann1.target[1]).result
-      anchor2 = am.createAnchor(ann2, ann2.target).result
-      anchor3 = am.createAnchor(ann3, ann2.target).result
-
-      hls = am.getHighlights [ann1, ann2]
-
-      assert.isArray hls
-      assert.include hls, anchor11.highlight[anchor11.startPage]
-      assert.include hls, anchor12.highlight[anchor12.startPage]
-      assert.include hls, anchor2.highlight[anchor2.startPage]
-      assert.notInclude hls, anchor3.highlight[anchor3.startPage]
-
-  describe 'Anchor.scrollIntoView()', ->
-    it 'calls scrollIntoView() on the highlight', ->
-      am = createAnchoringManager()
-      ann = createTestAnnotation "a1"
-      anchor = am.createAnchor(ann, ann.target[0]).result
-      anchor.scrollIntoView().then ->
-
-        assert.called anchor.highlight[anchor.startPage].scrollIntoView
+        assert.ok hl
+        assert.equal hl.page, page
+
+      it 'announces the creation of the highlights in an event', ->
+        am = createAnchoringManager()
+
+        assert.notCalled am.annotator.publish
```

Comment:
I wouldn't do negative testing like this. It will cause this test to fail unnecessarily if for whatever reason you later add an event publication when the anchoring manager is created.

If you really need to ensure that the spy is only called on `createAnchor`, then reset it immediately before the call:

```javascript
am.annotator.publish.reset()
am.createAnchor(...)
assert.calledWith(am.annotator.publish, ...)
```